strax/processing/peak_splitting.py

```diff
@@ -112,16 +153,20 @@ def _split_peaks(split_finder, peaks, orig_dt, is_split, min_area,
 
             prev_split_i = 0
             w = p['data'][:p['length']]
-
             for split_i, bonus_output in split_finder(
                     w, p['dt'], p_i, *args_options):
+                
+                # This is a bit odd here. Due tp the specific_outputs we have to get r 
+                # although we may not need it at all, but I do not see any nice way around
+                # this.
+                r = new_peaks[offset]
+                specific_output(r, p, split_i, bonus_output)
                 if split_i == NO_MORE_SPLITS:
-                    p['max_goodness_of_split'] = bonus_output
```

Comment:
why do you remove this line?

Reply:
I had to put this part into a separate function since hitlets do not have those fields. See line 89 - 105